pallets/multisig/src/lib.rs

```diff
@@ -533,6 +533,21 @@ pub mod pallet {
             pallet_identity::Module::<T>::join_identity(origin, auth_id)?;
             Ok(().into())
         }
+
+        /// Removes the admin identity from the `multisig`.  This must be called by the multisig itself.
+        #[pallet::call_index(17)]
+        #[pallet::weight(<T as Config>::WeightInfo::remove_admin())]
+        pub fn remove_admin(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+            let multisig = ensure_signed(origin)?;
+            let caller_did = Self::ensure_ms_has_did(&multisig)?;
+            let admin_did = AdminDid::<T>::take(&multisig);
+            Self::deposit_event(Event::MultiSigRemovedAdmin {
+                caller_did,
+                multisig,
+                admin_did,
```

Comment:
I think we should throw an error if there is no admin.  Then the event doesn't need to change.